frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -915,13 +915,18 @@ export const insightLogic = kea<insightLogicType>([
             actions.markInsightErrored(null)
             values.timeout && clearTimeout(values.timeout || undefined)
             const view = values.filters.insight
+            const capturedProps = props
             actions.setTimeout(
                 window.setTimeout(() => {
                     try {
-                        if (values && view == values.filters.insight) {
-                            actions.markInsightTimedOut(queryId)
+                        const mountedLogic = insightLogic.findMounted(capturedProps)
+                        if (!mountedLogic) {
+                            return
+                        }
+                        if (mountedLogic.values && view == mountedLogic.values.filters.insight) {
+                            mountedLogic.actions.markInsightTimedOut(queryId)
                             const tags = {
-                                insight: values.filters.insight,
+                                insight: mountedLogic.values.filters.insight,
                                 scene: sceneLogic.isMounted() ? sceneLogic.values.scene : null,
                             }
                             posthog.capture('insight timeout message shown', tags)
```

Comment:
Perhaps a simpler idea: save the `tags` json as an object above the timeout when you're sure everything is there and mounted... Then just capture that local variable.

Reply:
I think we throw on the `actions.blah` which I guess we can't capture?

Reply:
Hmm... good point :). `mountedLogic.values` should always exist though, so no need to be explicit with `mountedLogic.values && view == mountedLogic.values.filters.insight`... 🤔 

Here's another wildcard pattern that might work fine:

```js
listeners(() => ({
    startQuery: ({ queryId }, breakpoint) => {
          actions.setTimeout(
              window.setTimeout(() => {
                  try {
                      breakpoint()
                  } catch (e) {
                      // logic was either unmounted or `startQuery` was called again, abort timeout
                      return
                  }
              }
```

Reply:
have gone extra cautious and added both patterns but with a tombstone on the second try/catch block